esti/migrate_test.go

```diff
@@ -121,20 +138,86 @@ func createAndUploadParts(t *testing.T, logger logging.Logger, resp *s3.CreateMu
+	partsConcat, completedParts := createAndUploadParts(t, logger, &state.Multiparts.Info, migratePostPartsCount, migratePrePartsCount)
 
-	completeResponse, err := uploadMultipartComplete(svc, &state.Multipart.Info, append(state.Multipart.CompletedParts, completedParts...))
+	completeResponse, err := uploadMultipartComplete(svc, &state.Multiparts.Info, append(state.Multiparts.CompletedParts, completedParts...))
 	require.NoError(t, err, "failed to complete multipart upload")
 
 	logger.WithField("key", completeResponse.Key).Info("Completed multipart request successfully")
 
-	getResp, err := client.GetObjectWithResponse(ctx, state.Multipart.Repo, mainBranch, &api.GetObjectParams{Path: migrateMultipartFile})
+	getResp, err := client.GetObjectWithResponse(ctx, state.Multiparts.Repo, mainBranch, &api.GetObjectParams{Path: migrateMultipartsFile})
 	require.NoError(t, err, "failed to get object")
 	require.Equal(t, http.StatusOK, getResp.StatusCode())
 
-	preContentBytes, err := base64.StdEncoding.DecodeString(state.Multipart.Content)
+	preContentBytes, err := base64.StdEncoding.DecodeString(state.Multiparts.Content)
 	require.NoError(t, err, "failed to decode error")
 
 	fullContent := append(preContentBytes, partsConcat...)
 	require.Equal(t, fullContent, getResp.Body)
 }
+
+// Actions Tests
+
+func testPreMigrateActions(t *testing.T) {
+	// Create action data before migration
+	ctx, _, repo := setupTest(t)
+	state.Actions.Repo = repo
+	parseAndUploadActions(t, ctx, repo, mainBranch)
+	commitResp, err := client.CommitWithResponse(ctx, repo, mainBranch, &api.CommitParams{}, api.CommitJSONRequestBody{
+		Message: "Initial content",
+	})
+	require.NoError(t, err, "failed to commit initial content")
+	require.Equal(t, http.StatusCreated, commitResp.StatusCode())
+	_, err = responseWithTimeout(server, 1*time.Minute) // pre-commit action triggered on action upload, flush buffer
+	require.NoError(t, err)
+
+	testCommitMerge(t, ctx, repo)
+	testCreateDeleteBranch(t, ctx, repo)
+	testCreateDeleteTag(t, ctx, repo)
+
+	runResp := waitForListRepositoryRunsLen(ctx, t, repo, "", 13)
+	for _, run := range runResp.Results {
+		hookResp, err := client.ListRunHooksWithResponse(ctx, repo, run.RunId, &api.ListRunHooksParams{})
+		require.NoError(t, err, "failed to list runs")
+		require.Equal(t, http.StatusOK, hookResp.StatusCode())
+		ar := new(actionRun)
+		run.StartTime = run.StartTime.UTC()
+		*run.EndTime = run.EndTime.UTC()
+		ar.Run = run
+		ar.Hooks = hookResp.JSON200.Results
+		for i, task := range ar.Hooks {
+			ar.Hooks[i].StartTime = task.StartTime.UTC()
+			*ar.Hooks[i].EndTime = task.EndTime.UTC()
+		}
+		state.Actions.Runs = append(state.Actions.Runs, ar)
+	}
+}
+
+func testPostMigrateActions(t *testing.T) {
+	// Validate info using storage logs
+	ctx, _, _ := setupTest(t)
+	runs, err := client.ListRepositoryRunsWithResponse(ctx, state.Actions.Repo, &api.ListRepositoryRunsParams{})
```

Comment:
Can you trigger more actions from the Post test and see that the results are saved correctly with the old runs? For example, the ordering might get tricky.